pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/performance/UseStringBufferForStringAppendsTest.java

```diff
@@ -4,8 +4,33 @@
 
 package net.sourceforge.pmd.lang.java.rule.performance;
 
-import net.sourceforge.pmd.testframework.PmdRuleTst;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAllocationExpression;
+
+public class UseStringBufferForStringAppendsTest {
+    UseStringBufferForStringAppendsRule rule;
+
+    //CS304 (manually written) Issue link: https://github.com/pmd/pmd/issues/1736
+    @Before
+    public void init() {
+        rule = new UseStringBufferForStringAppendsRule();
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/pmd/pmd/issues/1736
+    @Test
+    public void visit() {
+        assertEquals(null, rule.visit(new ASTAllocationExpression(1), "yes"));
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/pmd/pmd/issues/1736
+    @Test(expected = NullPointerException.class)
+    public void visit2() {
+        ASTAllocationExpression e = null;
+        assertEquals("yes", rule.visit(e, "yes"));
+    }
 
-public class UseStringBufferForStringAppendsTest extends PmdRuleTst {
```

Comment:
The same comment applies here as well: Please re-enable the rule tests and add additional test cases to https://github.com/pmd/pmd/blob/master/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/performance/xml/UseStringBufferForStringAppends.xml